nifi-mock/src/test/java/org/apache/nifi/util/TestMockProcessSession.java

```diff
@@ -55,4 +55,35 @@ public void onTrigger(final ProcessContext ctx, final ProcessSession session) th
         }
 
     }
+
+
+    @Test(expected = AssertionError.class)
+    public void testNonExistentRelationFromProcessor() {
+        TestRunners.newTestRunner(NonExistentRelationTestProcessor.class).run();
+    }
+
+    protected static class NonExistentRelationTestProcessor extends PoorlyBehavedProcessor {
+
+            static final Relationship REL_NON_EXISTENT = new Relationship.Builder()
+                    .name("NonExistent Relation")
+                    .build();
+
+            public Set<Relationship> relationships = Collections.singleton(REL_FAILURE);
+
+            @Override
+            public Set<Relationship> getRelationships() {
+                return relationships;
```

Comment:
I think you just have to return an empty set of relationships instead of referencing the one of the other class. No?